src/components/tools/image-tool.tsx

```diff
@@ -84,6 +87,16 @@ export default class ImageToolComponent extends BaseComponent<IProps, IState> {
     this.imageDragDrop = new ImageDragDrop({
       isAcceptableImageDrag: this.isAcceptableImageDrag
     });
+
+    // Taken and modified from geometry-content.tsx.
+    // TODO: This should be abstracted for all tiles.
+    // if we haven't been assigned a title already, request one now
+    // we set the title without updating the content, so the title is ephemeral
+    if (!this.getContent().title) {
+      const { model: { id }, onRequestUniqueTitle } = this.props;
+      const title = onRequestUniqueTitle(id);
+      title && this.getContent().setTitle(title);
+    }
```

Comment:
When I ran this on a document with multiple image tiles, the result was that all image tiles received the same title: "Image 1". I believe this is because all of the components are instantiated in the first render of the document, which means they all request and receive the same title suggestion. Although this is not a problem created by any code you wrote -- it's really a limitation of the `onRequestUniqueTitle`/`getUniqueTitle` API -- it seems like we should consider it because this is going to be a common occurrence.

Reply:
The magic here happens in `getUniqueTitle()` in `src/models/document/document-content.ts`. This function basically looks at all tiles with names like "<TileType> <#>" and returns "<TileType> <Highest#+1>".

This is all happening in MobX State Tree, which I read up on but don't have much experience with yet. Is it possible that the first image is getting its title set to "Image 1", but when the second image is looking its title hasn't registered yet, so it thinks "Image 1" is still available? I'll play around a bit and see if I can find a solution.

Reply:
I made a first attempt at a solution to this problem. It doesn't work properly, but it offers useful insights. Long story short, I think the problem here is with how the title is being read in `ImageToolComponent`, which I'll try to address on Monday.

More detailed notes:
The solution is to add a global variable to `document-content.ts` which keeps track of the last number assigned and makes this the minimum when adding a new title.
- This solves the problem of loading a document with multiple untitled image tiles.
- The number is reused between image and graph tiles, which probably isn't desired but is actually helpful in revealing what I think the true problem is.
- Adding a new image tile skips a number using this solution (if the last image is 'Image 1', the new one will be 'Image 3'). It seems like the title is added when the tile is created, but the title is not visible in the constructor, `componentDidMount`, or `componentDidUpdate` (I tried all three), so it requests another unique title which bumps the number up again.
- This does _not_ happen when adding a geometry tile. In that case, the title assigned during creation is recognized so it doesn't try to add a new title and sticks with the next number in order.
- I suspect that loading a document with multiple untitled geometry tiles would result in different titles even without this global counter.
In conclusion, I think that the way I have the image tiles accessing their titles (`this.getContent().title`) is not working immediately, while it is in the geometry tile (`this.getContent().metadata.title`).

Now that I look over this one more time, I realize that `getTileTitle()` is probably failing in `getUniqueTitle()` in `src/models/document/document-content.ts` as well. I'll need to give this all more thought with a fresher mind later.